File: java/src/WarehouseActions/ReplenishStockCommand.java
package WarehouseActions;

import Ui.UserCommand;
import Ui.UserInput;
import Warehouse.Warehouse;

class ReplenishStockCommand implements UserCommand {

	private Warehouse warehouse;

	ReplenishStockCommand(Warehouse warehouse) {
		this.warehouse = warehouse;
	}

	@Override
	public void run(UserInput cmd) {
		if (cmd.args.length != 5) {
			System.err.println("ERROR: Usage: r aisle loc sku num");
			return;
		}
		String[] sku = new String[]{ cmd.args(1), cmd.args(2), cmd.args(3) };
		try {
			int numItems = Integer.parseInt(cmd.args(4));
			warehouse.replenish(sku, numItems);
		} catch (NumberFormatException e) {
			System.err.println("ERROR: Usage: r sku numitems");
		}
	}

}

File: csharp/WarehouseActions/ReplenishStockCommand.cs
namespace WarehouseActions {

	using System;
	using Products;
	using Ui;
	using Warehouse;

	class ReplenishStockCommand : UserCommand {

		private Warehouse warehouse;
		private Catalogue catalogue;

		internal ReplenishStockCommand(Warehouse warehouse, Catalogue catalogue) {
			this.warehouse = warehouse;
			this.catalogue = catalogue;
		}

		public void Run(UserInput cmd) {
			if (cmd.args.Length != 5) {
				Console.WriteLine("ERROR: Usage: r aisle loc sku num");
				return;
			}
			if (catalogue.Lookup(cmd.Args(3)) == null)
				throw new UnknownProductException(cmd.Args(3));
			string[] sku = new string[]{ cmd.Args(1), cmd.Args(2), cmd.Args(3) };
			try {
				int numItems = int.Parse(cmd.Args(4));
				warehouse.Replenish(sku, numItems);
			} catch (FormatException) {
				Console.WriteLine("ERROR: Usage: r sku numitems");
			}
		}

	}
}